python/tvm/driver/tvmc/frontends.py

```diff
@@ -0,0 +1,389 @@
+            raise TVMCException("input file not tflite")
+
+        if version != 3:
+            raise TVMCException("input file not tflite version 3")
+
+        logging.debug("tflite_input_type")
+        shape_dict, dtype_dict = TFLiteFrontend._input_type(tflite_model)
+
+        # parse TFLite model and convert into Relay computation graph
+        logging.debug("relay.frontend.from_tflite")
+        mod, params = relay.frontend.from_tflite(
+            tflite_model, shape_dict=shape_dict, dtype_dict=dtype_dict
+        )
+        return mod, params
+
+    @staticmethod
+    def _decode_type(n):
+        return TFLiteFrontend._tflite_m[n]
+
+    @staticmethod
+    def _input_type(model):
+        subgraph_count = model.SubgraphsLength()
+        assert subgraph_count > 0
+        shape_dict = {}
+        dtype_dict = {}
+        for subgraph_index in range(subgraph_count):
+            subgraph = model.Subgraphs(subgraph_index)
+            inputs_count = subgraph.InputsLength()
+            assert inputs_count >= 1
+            for input_index in range(inputs_count):
+                input_ = subgraph.Inputs(input_index)
+                assert subgraph.TensorsLength() > input_
+                tensor = subgraph.Tensors(input_)
+                input_shape = tuple(tensor.ShapeAsNumpy())
+                tensor_type = tensor.Type()
+                input_name = tensor.Name().decode("utf8")
+                shape_dict[input_name] = input_shape
+                dtype_dict[input_name] = TFLiteFrontend._decode_type(tensor_type)
+
+        return shape_dict, dtype_dict
+
+
+class PyTorchFrontend(Frontend):
+    """ PyTorch frontend for TVMC """
+
+    @staticmethod
+    def name():
+        return "pytorch"
+
+    @staticmethod
+    def suffixes():
+        # Torch Script is a zip file, but can be named pth
+        return ["pth", "zip"]
+
+    def load(self, path, shapes):
+        # pylint: disable=C0415
+        import torch
+        from tvm import relay
+
+        if not shapes:
+            raise TVMCException(
+                "--input-shape must be specified for {}".format(self.name())
+            )
+
+        traced_model = torch.jit.load(path)
+        traced_model.eval()  # Switch to inference mode
+        input_shapes = [
+            ("input{}".format(idx), shape) for idx, shape in enumerate(shapes)
+        ]
+        logging.debug("relay.frontend.from_pytorch")
+        return relay.frontend.from_pytorch(traced_model, input_shapes)
+
+
+ALL_FRONTENDS = [
+    KerasFrontend,
+    OnnxFrontend,
+    TensorflowFrontend,
+    TFLiteFrontend,
+    PyTorchFrontend,
+]
+
+
+def get_frontends():
+    """Return the names of all supported frontends"""
+    return [frontend.name() for frontend in ALL_FRONTENDS]
+
+
+def lookup_frontend(name):
+    for frontend in ALL_FRONTENDS:
+        if name == frontend.name():
+            return frontend()
+    raise TVMCException("unrecognized frontend")
+
+
+def guess_input_language(path):
```

Comment:
- We are not using "language" anymore.
- By its name, I don't expect this function to return a frontend but should just a frontend name; otherwise you may need to rename it to something like `guess_frontend`.

Reply:
Done